hazelcast/src/main/java/com/hazelcast/concurrent/lock/LockServiceImpl.java

```diff
@@ -328,11 +332,21 @@ private void clearLockStoresHavingLesserBackupCountThan(int partitionId, int thr
 
     @Override
     public DistributedObject createDistributedObject(String objectId) {
-        return new LockProxy(nodeEngine, this, objectId);
+        LockProxy lockProxy = new LockProxy(nodeEngine, this, objectId);
+
+        if (logger.isFinestEnabled()) {
+            logger.finest("Created lock " + objectId);
```

Comment:
IMO, we should talk only about a *"LockProxy"* here and not the *"lock"*.

If I remember correctly, the proxy can exist without the real object in the cluster.

Reply:
That's right, the proxy merely sends operations and is agnostic as to the existence of the actual lock container